include/dca/phys/dca_algorithms/compute_free_greens_function.hpp

```diff
@@ -74,18 +76,24 @@ void compute_G0_k_t(
     clustermapping::quadrature_integration<Scalar, KDmn, OrbitalSpinDmn>::quadrature_integration_G_q_t_st(
         beta, sign, tau, mu_function, H0_k, g);
 
-    for (int k = 0; k < KDmn::dmn_size(); ++k) {
-      for (int n = 0; n < OrbitalSpinDmn::dmn_size(); ++n) {
-        for (int m = 0; m < OrbitalSpinDmn::dmn_size(); ++m) {
-          if (g(m, n, k).imag() > 1.e-6)
-            throw std::logic_error("G_0(\vec{k}, \tau) is real!");
-          G0_k_t(m, n, k, t) = g(m, n, k).real();
-        }
-      }
-    }
+    std::copy_n(g.values(), g.size(), &G0_k_t(0, 0, 0, t));
```

Comment:
Taking the address operator on G0_k_t fills me with suspicion, I think this make assumptions about the internal ordering of both functions which is quite suspect.  Refactoring in func::function could easily break this, operator() is returning a reference to a scalar_type no guarantee

Reply:
Also there needs to be a copy_n test for func::function that will break as soon as this assumption is violated.

Reply:
The code relies on functions and matrices having column major order. For example include/dca/phys/dca_step/cluster_solver/ctaux/ctaux_cluster_solver.hpp:566. I will add a test to fix the ordering.

Reply:
At some point it would be good if we did the copy using our own operator constructor or whatever so internal layout wasn't exposed like this.  Not for this PR.

Reply:
Tested  in 9f7aa28c694fb19450f647fe4dc1df02010bf5e2.
I guess it would be good, but then this is very general and works for any container without having to specialize for Vector, Matrix, some other sort of Matrix, and std::vector. I am not sure we want to remove raw pointers from modern high-performance code, but we can discuss about this later.